archive/m/modula2/FractionMath.mod

```diff
@@ -0,0 +1,213 @@
+        RETURN TRUE;
+    ELSE
+        RETURN FALSE;
+    END;
+END greaterOrEqual;
+
+PROCEDURE [<=] lessOrEqual(f1, f2 : Fraction) : BOOL;
+BEGIN
+    IF (f1.numerator * f2.denominator) <= (f1.denominator * f2.numerator) THEN
+        RETURN TRUE;
+    ELSE
+        RETURN FALSE;
+    END;
+END lessOrEqual;
+
+PROCEDURE [+] Add(f1, f2 : Fraction) : Fraction;
+VAR fResult : Fraction;
+BEGIN
+    CREATE(fResult);
+    IF f1.denominator = f2.denominator THEN
+        fResult.SetFraction(f1.numerator + f2.numerator, f1.denominator);        
+    ELSE
+        fResult.SetFraction((f1.numerator * f2.denominator) + (f2.numerator * f1.denominator), f1.denominator * f2.denominator);
+        gcd = GreatestCommonDivisor(fResult.numerator, fResult.denominator);
+        fResult.SetFraction(fResult.numerator / gcd, fResult.denominator / gcd);
+    END;
+END Add;
+
+PROCEDURE [-] Sub(f1, f2 : Fraction) : Fraction;
+VAR fResult : Fraction;
+BEGIN
+    CREATE(fResult);
+    IF f1.denominator = f2.denominator THEN
+        fResult.SetFraction(f1.numerator - f2.numerator, f1.denominator);        
+    ELSE
+        fResult.SetFraction((f1.numerator * f2.denominator) - (f2.numerator * f1.denominator), f1.denominator * f2.denominator);
+        gcd = GreatestCommonDivisor(fResult.numerator, fResult.denominator);
+        fResult.SetFraction(fResult.numerator / gcd, fResult.denominator / gcd);
+    END;
+END Sub;
+
+PROCEDURE [*] Mult(f1, f2 : Fraction) : Fraction;
+VAR fResult : Fraction;
+BEGIN
+    CREATE(fResult);
+    fResult.SetFraction(f1.numerator * f2.numerator, f1.denominator * f2.denominator);
+    gcd = GreatestCommonDivisor(fResult.numerator, fResult.denominator);
+    fResult.SetFraction(fResult.numerator / gcd, fResult.denominator / gcd);
+END Mult;
+
+PROCEDURE [/] Div(f1, f2 : Fraction) : Fraction;
+VAR fResult : Fraction;
+BEGIN
+    CREATE(fResult);
+    fResult.SetFraction(f1.numerator * f2.denominator, f1.denominator * f1.numerator);
+    gcd = GreatestCommonDivisor(fResult.numerator, fResult.denominator);
+    fResult.SetFraction(fResult.numerator / gcd, fResult.denominator / gcd);
+END Div;
+
+BEGIN 
+    SetFraction(1, 1);
+END Fraction;
+
+
+VAR fracOne : Fraction;
+    fracTwo : Fraction;
+      
+PROCEDURE result(f1 : INTEGER, op : STRING, f2 : INTEGER) : Fraction
+BEGIN
+    IF op = "+" THEN 
+        RETURN f1 + f2;
+    ELSE IF op = "-" THEN 
+        RETURN f1 - f2;
+    ELSE IF op = "*" THEN 
+        RETURN f1 * f2;
+    ELSE IF op = "/" THEN 
+        RETURN f1 / f2;
+    ELSE IF op = ">" THEN 
+        RETURN f1 > f2
+    ELSE IF op = ">=" THEN 
+        RETURN f1 >= f2
+    ELSE IF op = "<" THEN 
+        RETURN f1 < f2
+    ELSE IF op = "<=" THEN 
+        RETURN f1 <= f2
+    ELSE IF op = "=" THEN 
+        RETURN f1 = f2
+    ELSE IF op = "#" THEN 
+        RETURN f1 # f2
+    ELSE
+        RETURN WriteString("Invalid operator");
+    END;    
+END result;
+
+BEGIN 
```

Comment:
Command line arguments are used for inputs, and stdout is used for outputs for all samples. Please see the [project requirements](https://sampleprograms.io/projects/fraction-math/) for details.